tests/integration/test_spans.py

```diff
@@ -1195,6 +1207,41 @@ def test_span_ingestion(
     metrics_consumer.assert_empty()
 
 
+def assert_contains(span, expected_span):
+    """Assert that an old-style kafka span is contained within the new backward compatible span.
+
+    This function can be removed once the consumer uses the new fields."""
+
+    # These keys are produced by the old producer, but we chose not to replicate them because
+    # the consumer does not need them anymore:
+    unused_keys = {
+        "exclusive_time_ms",
+        "is_segment",
+        "measurements",
+        "tags",
+        "sentry_tags",
+    }
+
+    # These keys were set unconditionally by the store serializer, but
+    # can be omitted if False:
+    optional_flags = {"is_remote"}
+
+    for key, expected_value in expected_span.items():
+        if key in unused_keys:
+            continue
+        if key == "data":
+            # Data may have more fields
+            for key in expected_value:
+                assert span["data"][key] == expected_value[key]
+        else:
+            if key in optional_flags:
+                assert span.get(key, False) == expected_span[key]
+            elif key == "links":
+                continue  # FIXME
```

Comment:
Should not merge until we know for sure that we can produce link attributes using the new format.